symengine/parser.h

```diff
@@ -0,0 +1,17 @@
+#ifndef SYMENGINE_PARSER_H
+#define SYMENGINE_PARSER_H
+
+#include <symengine/basic.h>
+#include <symengine/dict.h>
+#include <symengine/parser.cpp>
```

Comment:
You should not include a `.cpp` file. Remove this include and just change
```c++
inline RCP<const Basic> parse(std::string& s) {
	ExpressionParser p;
	return p.parse_expr(s);
} 
```
to
```c++
RCP<const Basic> parse(const std::string& s);
```
and implement it in `parser.cpp`. Also make the input argument `const`, as I did.